stonesoup/hypothesiser/probability.py

```diff
@@ -126,18 +146,57 @@ def hypothesise(self, track, detections, timestamp, **kwargs):
                 prediction, detection.measurement_model, **kwargs)
             # Calculate difference before to handle custom types (mean defaults to zero)
             # This is required as log pdf coverts arrays to floats
-            log_pdf = mn.logpdf(
+            log_pdf = multivariate_normal.logpdf(
                 (detection.state_vector - measurement_prediction.state_vector).ravel(),
                 cov=measurement_prediction.covar)
             pdf = Probability(log_pdf, log_value=True)
-            probability = (pdf * self.prob_detect)/self.clutter_spatial_density
 
-            # True detection hypothesis
-            hypotheses.append(
-                SingleProbabilityHypothesis(
-                    prediction,
-                    detection,
-                    probability,
-                    measurement_prediction))
+            if self._is_valid_measurement(measurement_prediction, detection):
+                validated_measurements += 1
+                valid_measurement = True
+
+            if self.include_all or valid_measurement:
+                probability = pdf * self.prob_detect
+                if self.clutter_spatial_density is None:
+                    # Note: will divide by validated measurements count later...
+                    probability *= self._validation_region_volume(
+                        self.prob_gate, measurement_prediction)
+                else:
+                    probability /= self.clutter_spatial_density
```

Comment:
```suggestion
                if self.clutter_spatial_density is not None:
                    probability /= self.clutter_spatial_density
```

Might be more readable if the calculation for the case ``self.clutter_spatial_density is None`` is all done in one place on lines 175-177, given that we are iterating over the hypotheses and making the check anyway. The comment is helpful, but I think keeping the "equations" together is more helpful.